tests/commands/test_init.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2020 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For further info, check https://github.com/canonical/charmcraft
+
+from argparse import Namespace
+import subprocess
+
+from charmcraft.commands.init import InitCommand
+from charmcraft.commands._utils import S_IXALL
+from tests.test_infra import pep8_test, get_python_filepaths
+
+
+def test_init_pep8(tmp_path, *, name=None, author="J Doe"):
+    if name is None:
+        name = tmp_path.name
+    cmd = InitCommand('group')
+    cmd.run(Namespace(directory=tmp_path, name=name, author=author))
+    paths = get_python_filepaths(
+        roots=[str(tmp_path / "src"), str(tmp_path / "tests")],
+        python_paths=[])
+    pep8_test(paths)
+
+
+def test_init_non_ascii_author(tmp_path):
+    test_init_pep8(tmp_path, author="فلانة الفلانية")
+
+
+def test_init_non_ascii_name(tmp_path):
+    test_init_pep8(tmp_path, name="ჭიჭიკო")
```

Comment:
fwiw, I know Juju expressly forbids nonascii charm names, I'm not sure if the charm store does/doesn't want to support nonascii elsewhere.

Reply:
The Store is surely opinionated about this (it's probably a subset of ascii, like a-z, A-Z, 0-9, -, and not much more...

Reply:
The [docs] say `^[a-z][a-z0-9-]*[a-z]$`, pretty much

[docs]: https://discourse.juju.is/t/charm-metadata/1043

Reply:
I've added a test for this